packages/yarnpkg-core/sources/Manifest.ts

```diff
@@ -371,7 +376,8 @@ export class Manifest {
       }
     }
 
-    return errors;
+    // @ts-ignore: It's ok to initialize it now
+    this.errors = errors;
```

Comment:
I think you don't need the `@ts-ignore` if you remove the `readonly` attribute

Reply:
This is used with the (seemingly) same intent in other instances in this codebase: private write, public readonly. Typescript doesn't support this right now. public ReadonlyArray is the closest thing and doesn't require `@ts-ignore`.

Reply:
Aaah yes, right, I forgot about this quirk 🙁